docs/polkadot/networking/3-avail-valid.md

```diff
@@ -230,57 +233,42 @@ When receiving each piece, we also need to be able to authenticate it individual
 
-Then, we also need a proxying protocol that allows target peer T to request piece X from source peer S via a proxy peer P, which ideally should be spam-resistant (TODO [TBX S1 #3]). Proxies may optionally store the pieces that they are passed, and broadcast receipts for these as well, up to some limit chosen by them to conserve their own resources.
+(a) was the assumption we made of all nodes in the main proposal, and now we must also account for (b). Note that this is a more restricted assumption than an arbitrary internet topology - the latter would require a fully-general NAT traversal solution, which is more complex and carries more runtime overhead.
 
-**Distributors**
+Specifically for A&V direct sending, this translates to the following scenarios:
 
-For a distributor (c, i), when sending a piece to another target t, we augment the sending with some failover proxies. If the direct sending fails after a timeout, we proceed as follows, trying successive items with a timeout until one succeeds:
+a. for incoming connections, the sentry node accepts these and proxies them back to the validator node<br/>
+b. for outgoing connections, either the validator node makes the connection directly, or else makes it via their sentry node.
 
-2. [TBX S1 #2] send to t, via randomly-chosen proxies p from the following sets, in turn:
+In some cases where both peers are behind their own sentries, this may be up to 2 hops. However, it is unnecessary to have special-case logic to handle this situation. The following general rules will suffice, and they can be applied even to normal validator nodes (those running without sentries):
 
-   a. your (and their) ring, with co-ords (c', i) for all c'
-   b. their parachain, with co-ords (c, i') for all i'
-   c. the gossip neighbours of t
+1. For the address book, the validator should insert (or have their sentry nodes insert) the addresses of whatever nodes are acting as incoming proxies for it, what other people can reach. The following details are important:
 
-   k choices are to be tried from each set, before moving onto the next set; k = 3 seems like a reasonable *a priori* choice without real-world data.
+    a. each record should include a creation date, so that later entries unambiguously obsolete earlier entries. Thus load-balancing can be done in a more predictable way, across the full set of addresses.
 
-   The choice should prioritise proxies with known shared reachability to t.
+    b. each record should include an expiry date, so that old addresses are unambiguously avoided by readers unsure if the entry they got is "too old" or not
 
-A proxy has "known shared reachability" to a target t iff:
+2. For transport session keys, any node claiming to be a validator or a proxy for one, must present a certificate proving that the validator key authorises the transport key to do so. This ties back into the [authentication proposals](#L-authentication.html#proposal-fresh-authentication-signals).
```

Comment:
These *certificates* would tie into the proposal in https://github.com/w3f/research/blob/master/docs/polkadot/networking/L-authentication.rst correct?

Reply:
Yes, there is a link right there :)